src/unix-manager.c

```diff
@@ -700,18 +702,112 @@ TmEcode UnixManagerCaptureModeCommand(json_t *cmd,
     SCReturnInt(TM_ECODE_OK);
 }
 
-TmEcode UnixManagerReloadRules(json_t *cmd, json_t *server_msg, void *data)
+TmEcode UnixManagerReloadRulesWrapper(json_t *cmd, json_t *server_msg, void *data, int wait)
 {
     SCEnter();
-    DetectEngineReloadStart();
+    int r = DetectEngineReloadStart();
 
-    while (DetectEngineReloadIsDone() == 0)
-        usleep(100);
+    if (wait) {
+        while (DetectEngineReloadIsDone() == 0)
+            usleep(100);
+    } else {
+        if (DetectEngineReloadIsDone() == 0 && r == -1) {
```

Comment:
Ok that's right, but when we're not waiting detect_sync.state will stay to RELOAD and never set to IDLE (DetectEngineReloadIsDone).
It won't be possible to redo another rule reload.